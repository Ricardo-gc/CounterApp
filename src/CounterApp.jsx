
export const CounterApp = ( { value } ) => {

    const handleApp = (event, newValue) => {
        console.log(event);
    }
    
    return (
        <>
            <h1>Counter App</h1>
            <h2>{ value }</h2>

            <button onClick={ handleApp }>
                +1    
            </button>
        </>
    );
}