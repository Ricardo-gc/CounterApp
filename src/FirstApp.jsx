import React from 'react'
import PropTypes from 'prop-types';

export const FirstApp = ( {
  title, 
  subTitle,
  name
} ) => {
  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <p>{ subTitle }</p>
      <div>{ name }</div>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

FirstApp.defaultProps = {
  title: 'No title',
  subTitle: 'No SubTitle',
  name: 'Ricardo Garc√≠a'
}

