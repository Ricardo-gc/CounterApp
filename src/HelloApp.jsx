// import React from 'react';

// function App(){
//     // documento.createElement.....
//     return ( <h1>Hello World!!</h1> );
// }

// export default App;
//rafc
import React from 'react'

export const HelloWorldApp = () => {
  return (
    <h1>Hello World App</h1>
  );
}

