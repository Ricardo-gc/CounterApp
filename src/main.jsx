import React from 'react';
import ReactDOM from 'react-dom/client';

import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloApp';
import { CounterApp } from './CounterApp';

// Para importarlo de manera global
import './index.css';

//render el component
ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <HelloWorldApp /> */}
    {/* <FirstApp title='Hello World!!'/> */}
    <CounterApp value={ 7 }/>
  </React.StrictMode>
)
