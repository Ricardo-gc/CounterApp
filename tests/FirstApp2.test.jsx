import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas es FirstApp', () => {
    const title = 'Hola Mundo';
    const subTitle = 'Ich bin eine Subtitle';
    test('Shouel match with snapshot', () => {
        const {container} = render (<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('Should show the message: Hola Mundo', () => {
        render (<FirstApp title={title}/>);
        expect(screen.getByText(title)).toBeTruthy();
        //screen.debug();
    });
    
    test('Should show title in a h1 tag', () => {
        render (<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('Should show subtitle send by props', () => {
        render (<FirstApp title={title} subTitle={ subTitle }/>);
        expect(screen.getByText(subTitle)).toBeTruthy();
        //expect(screen.getAllByAltText(subTitle).length).toBe(2);
    });
});