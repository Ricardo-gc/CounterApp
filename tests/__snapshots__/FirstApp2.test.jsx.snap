// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas es FirstApp Shouel match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola Mundo
  </h1>
  <p>
    No SubTitle
  </p>
  <div>
    Ricardo Garc√≠a
  </div>
</div>
`;
